### PR TITLE
[metrics] Standardize metrics.Count API to prometheus counter

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

- Rename Count -> Counter.
- Expose `.inc()` method instead of `.record()`.
- Do some validation.

Still supports Count but logs a deprecation warning (will remove next release).

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
FYI @thomasdesr

**Review by rkooo567 — APPROVED** [review]
This is a pretty breaking change? But since we didn't make a doc yet, it might be fine haha...

**Comment by edoakes** [comment]
@rkooo567 it's a smooth deprecation - the old syntax is still supported

File: python/ray/util/metrics.py [hunk 1]
```diff
@@ -149,6 +150,43 @@ def __reduce__(self):
         serialized_data = (self._name, self._description, self._tag_keys)
```
Comment (franklsf95) [inline python/ray/util/metrics.py:150]: L149 should the deserializer be renamed to `Counter` too? I'm getting a lot of warnings when pickling the Counter objects to remote tasks:

```
WARNING metrics.py:206 -- `metrics.Count` has been renamed to `metrics.Counter`. `metrics.Count` will be removed in a future release.
```

**Review by frank-lsf — COMMENTED** [review]
cc @edoakes

File: python/ray/util/metrics.py (hunk 1 above)
Comment (edoakes) [reply python/ray/util/metrics.py:150]: Oh, good catch @franklsf95! Any chance you would like to make a PR fixing it?

**Pull request merged**